src/features/filter/reducers/toggleClassFilterAll.thunk.ts

```diff
@@ -0,0 +1,50 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { filterAdapter, simpleSelectFilterById } from "../filter.selectors";
+import { RootState } from "../../../app/store/store";
+import { FilterKey, ClassFilterValue, Filter } from "../types/filter.types";
+import { jdiClassFilterInit } from "../utils/filterSet";
+import { PageObjectId } from "../../pageObjects/types/pageObjectSlice.types";
+import { ElementLibrary } from "../../locators/types/generationClasses.types";
+import { LocalStorageKey, setLocalStorage, getLocalStorage } from "../../../common/utils/localStorage";
+
+export const toggleClassFilterAll = createAsyncThunk(
+  "filter/toggleClassFilterAll",
+  async (payload: { pageObjectId: PageObjectId; library: ElementLibrary; value: boolean }, { getState }) => {
+    const { pageObjectId, value, library } = payload;
+    const state = getState() as RootState;
+    let newFilterValue = simpleSelectFilterById(state.filters, pageObjectId);
+
+    if (!newFilterValue) {
+      newFilterValue = { pageObjectId, [FilterKey.JDIclassFilter]: jdiClassFilterInit(library) };
+    }
+    const newFilter = { ...newFilterValue[FilterKey.JDIclassFilter] };
+    Object.keys(newFilter).forEach((key: string) => {
+      // don't know how to fix it
+      //@ts-ignore
+      newFilter[key] = value;
+    });
+
+    if (!getLocalStorage(LocalStorageKey.Filter)) {
+      setLocalStorage(LocalStorageKey.Filter, { [library]: newFilter });
+    } else {
+      const savedFilters = getLocalStorage(LocalStorageKey.Filter);
```

Comment:
instead of evaluate `getLocalStorage` twice, it's better to evaluate it before `if`